### PR TITLE
Add some examples

### PR DESCRIPTION
**Pull request by ouyuanning**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

**Review by zengyan1 — APPROVED** [review]

File: pkg/sql/colexec/evalProjection.go [hunk 1]
```diff
@@ -57,49 +47,46 @@ func (projection *Projection) EvalProjection(bat *batch.Batch, proc *process.Pro
 	if bat == nil || bat.IsEmpty() || bat.Last() {
```
Comment (fengttt) [inline pkg/sql/colexec/evalProjection.go:47]: This test, bat == nil || bat.IsEmpty() || bat.Last() should be wrapped in one method.    Who knows what is bat.Last()?   What is the difference between Last() and Empty(), etc etc.

File: pkg/sql/colexec/evalProjection.go [hunk 2]
```diff
@@ -57,49 +47,46 @@ func (projection *Projection) EvalProjection(bat *batch.Batch, proc *process.Pro
...
 		return bat, nil
 	}
+	if len(projection.ProjectExecutors) == 0 {
+		return bat, nil
```
Comment (fengttt) [inline pkg/sql/colexec/evalProjection.go:51]: Does project zero columns means no project at all?

File: pkg/sql/colexec/evalProjection.go [hunk 3]
```diff
@@ -57,49 +47,46 @@ func (projection *Projection) EvalProjection(bat *batch.Batch, proc *process.Pro
...
-						if oldV != nil && newV == oldV {
-							bat.Vecs[k] = nil
-						}
-					}
-				}
-			}
-			projection.projectBat = nil
 			return nil, err
 		}
 		projection.projectBat.Vecs[i] = vec
```
Comment (fengttt) [inline pkg/sql/colexec/evalProjection.go:61]: This line worth a comment, on why we can just assign.    I believe many later code will copy what you did here, they better understand what is going on.

File: pkg/sql/colexec/evalProjection.go [hunk 4]
```diff
@@ -57,49 +47,46 @@ func (projection *Projection) EvalProjection(bat *batch.Batch, proc *process.Pro
...
 	}
-	newAlloc, err := FixProjectionResult(proc, projection.ProjectExecutors, projection.uafs, projection.projectBat, bat)
-	if err != nil {
-		return nil, err
-	}
-	projection.ProjectAllocSize = max(projection.ProjectAllocSize, int64(newAlloc))
+	projection.ProjectAllocSize = max(projection.ProjectAllocSize, int64(projection.projectBat.Size()))
```
Comment (fengttt) [inline pkg/sql/colexec/evalProjection.go:63]: This is not necessarily true -- all projected columns does not allocate anything.    But fine, ..., less important.

**Review by fengttt — COMMENTED** [review]
So ZengYan's change has been merged into our working branch?

File: pkg/sql/colexec/evalProjection.go (hunk 2 above)
Comment (ouyuanning) [reply pkg/sql/colexec/evalProjection.go:51]: 这个是给所有算子用的。比如 table_scan，有的时候会没有projectionList，这个时候就会是0

File: pkg/sql/colexec/evalProjection.go (hunk 4 above)
Comment (ouyuanning) [reply pkg/sql/colexec/evalProjection.go:63]: 要想精确其实有点麻烦。对于列表达式可能没有alloc， 但是对于比如  a+1，  其实是有的，只是它只会alloc一次。
另外因为行数不同，每个批次所占用的size会有差异，比如第一批次1行，第二批次8000行，那么我们以8000行的那个结果给统计信息

**Comment by ouyuanning** [comment]
> So ZengYan's change has been merged into our working branch?

是的

File: pkg/sql/colexec/evalProjection.go (hunk 3 above)
Comment (ouyuanning) [reply pkg/sql/colexec/evalProjection.go:61]: 好的

File: pkg/sql/colexec/evalProjection.go (hunk 1 above)
Comment (ouyuanning) [reply pkg/sql/colexec/evalProjection.go:47]: 这个我没有提取是因为
1、对于bat.IsEmpty，不同算子它有不同的处理方式。感觉正常应该是类似continue，直接继续call一次child
     而对于nil来说，应该是整个要结束。所以我理解他们语义是不一样的。如果提取到类似 Done() 的函数，会把这两个语义的不一致给抹掉。感觉有点危险

2、Last(）这个我也不知道是啥情况，
所以我这个PR就先没有动它

Comment (ouyuanning) [reply pkg/sql/colexec/evalProjection.go:47]: Empty()只是说明上一个算子传递过来的数据长度是0，如果当前算子对于这种情况没有特殊处理需求，那应该继续执行children.Call，再拿下一批次数据应该就可以了，没必要往后传

**Review by badboynt1 — APPROVED** [review]

**Pull request merged**